backend/src/routes/api/storage/index.ts

```diff
@@ -0,0 +1,63 @@
+import { FastifyInstance, FastifyReply } from 'fastify';
+import { createMinioClient, getObjectStream } from './storageUtils';
+import { getDashboardConfig } from '../../../utils/resourceUtils';
+import { getDirectCallOptions } from '../../../utils/directCallUtils';
+import { getAccessToken } from '../../../utils/directCallUtils';
+import { OauthFastifyRequest } from '../../../types';
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  const dashConfig = getDashboardConfig();
+  if (dashConfig?.spec.dashboardConfig.disableS3Endpoint === false) {
+    fastify.get(
+      '/:namespace/:bucket',
+      async (
+        request: OauthFastifyRequest<{
+          Querystring: Record<string, string>;
+          Params: { '*': string; [key: string]: string };
+          Body: { [key: string]: unknown };
+        }>,
+        reply: FastifyReply,
+      ) => {
+        try {
+          const { namespace, bucket } = request.params;
+          const query = request.query;
+          const key = query.key;
+
+          const requestOptions = await getDirectCallOptions(fastify, request, request.url);
+          const token = getAccessToken(requestOptions);
+
+          const stream = await getObjectStream({
+            bucket,
+            client: await createMinioClient(fastify, token, namespace),
+            key,
+          });
+
+          reply.type('text/plain');
+
+          await new Promise<void>((resolve, reject) => {
+            stream.on('data', (chunk) => {
+              reply.raw.write(chunk);
+            });
+
+            stream.on('end', () => {
+              reply.raw.end();
+              resolve();
+            });
+
+            stream.on('error', (err) => {
+              fastify.log.error('Stream error:', err);
+              reply.raw.statusCode = 500;
+              reply.raw.end(err.message);
+              reject(err);
+            });
+          });
```

Comment:
My leading concern -- which I'm not sure we need to address it here -- but should before we release it. 

If the resource you're pulling is so large the data crashes our pod, it's not going to be awesome. You're also holding up the connection until you're done -- which may time out.

We should probably find a way to peek only the first "bunch" of characters. Are we responsible for the "full download / view in a new tab"?

Reply:
It would appear we are on the hook for the new tab too... if my scanning of the code was accurate

Reply:
As far as i could tell, the KF ui will attempt to pull all in the new tab unless told to preview